### DISCUSSION TITLE
Feature request: [react-instantsearch-nextjs] NextJS v15 support

### DISCUSSION THREAD

**Discussion by wesleyjanse** [algolia/instantsearch#6490] [Ideas] [answered]
I would like to see Next 15 support by making sure that the headers accessed in react-instantsearch-nextjs are awaited.

**Comment by Haroenv** [comment]
We plan to work on this (and the other stumbling blocks for Next) in the next year. In the mean time if you have an approach to enable this, you could create a patch with the needed changes (make the function asynchronous mainly)

**Comment by wesleyjanse** [comment, marked as answer]
Fixed with versions:

    "react-instantsearch": "7.14",
    "react-instantsearch-nextjs": "0.4.0",
